docs/config/server-options.md

```diff
@@ -95,13 +95,13 @@ export default defineConfig({
       '/api': {
         target: 'http://jsonplaceholder.typicode.com',
         changeOrigin: true,
-        rewrite: (path) => path.replace(/^\/api/, '')
+        rewrite: (path) => path.replace('/^\/api/', '')
```

Comment:
Ahh, that did work! Sorry, everything I found online included the rewrite, but it just was not working.

I came to the docs and came to the same conclusion since it was in options and regex.

Maybe it could be made a little more clear that proxy matches following routes (i.e. '/api/products', /api/products/1', etc.) by default using:  
```proxy: {
      '/api': {
        target: 'http://localhost:5000',
        changeOrigin: true,
        secure: false,
      }```
      
      It was confusing for me at least.

Reply:
Yeah, I think we could have examples.
```
// string shorthand: http://localhost:5173/foo -> http://localhost:4567/foo
// with options: http://localhost:5173/api/bar-> http://jsonplaceholder.typicode.com/bar
```

Reply:
for readability, I was thinking something more along the lines of:
```js
export default defineConfig({
  server: {
    proxy: {
      // string shorthand, matches calls to '/foo/bar' by default
      '/foo': 'http://localhost:4567',
      // with options, returns http://jsonplaceholder.typicode.com/bar
      '/api': {
        target: 'http://jsonplaceholder.typicode.com',
        changeOrigin: true,
        rewrite: (path) => path.replace(/^\/api/, '')
      },```

Reply:
the matching subsequent folders by default is what was confusing me. I thought I needed a rewrite to achieve this.